code/game/objects/items.dm

```diff
@@ -490,22 +490,54 @@ modules/mob/living/carbon/human/life.dm if you die, you will be zoomed out.
 			fail_chance += round(H.shock_stage/120 * 40)
 			base_time += round(H.shock_stage/120 * 40)
 
+	//Precalculate worktime here
+	var/time_to_finish = 0
+	if (base_time)
+		time_to_finish = base_time - get_tool_quality(required_quality) - user.stats.getStat(required_stat)
+
+	if((instant_finish_tier < get_tool_quality(required_quality)))
+		time_to_finish = 0
+
+
+	//Repeating sound code!
+	//A datum/repeating_sound is a little object we can use to make a sound repeat a few times
+	var/datum/repeating_sound/toolsound = null
 	if(forced_sound != NO_WORKSOUND)
+
+		var/soundfile
 		if(forced_sound)
-			playsound(src.loc, forced_sound, 100, 1)
+			soundfile = forced_sound
 		else
-			playsound(src.loc, src.worksound, 100, 1)
+			soundfile = worksound
 
-	if(base_time && (instant_finish_tier >= get_tool_quality(required_quality)))
+		if (sound_repeat && time_to_finish)
+			//It will repeat roughly every 2.5 seconds until our tool finishes
+			toolsound = new/datum/repeating_sound(sound_repeat,time_to_finish,0.15, src, soundfile, 80, 1)
+		else
+			playsound(src.loc, soundfile, 100, 1)
+
+	//The we handle the doafter for the tool usage
+	if(time_to_finish)
 		target.used_now = TRUE
-		var/time_to_finish = base_time - get_tool_quality(required_quality) - user.stats.getStat(required_stat)
+
 		if(!do_after(user, time_to_finish, user))
+			//If the doafter fails
 			user << SPAN_WARNING("You need to stand still to finish the task properly!")
 			target.used_now = FALSE
+			if (toolsound)
+				//We abort the repeating sound.
+				//Stop function will delete itself too
+				toolsound.stop()
+				toolsound = null
 			return TOOL_USE_CANCEL
 		else
 			target.used_now = FALSE
 
+	//Safe cleanup
+	if (toolsound)
+		toolsound.stop()
+	toolsound = null
```

Comment:
Why do we null it even if it's already null?

Reply:
incase it's not already null
to ensure it can GC cleanly

Reply:
Don't we... Uh. Check if it's null one line above?